src/rand.jl

```diff
@@ -0,0 +1,70 @@
+function Random.rand(rng::AbstractRNG, ::Random.SamplerType{R}) where R <: Union{<:Rotation{2},<:RotMatrix{2}}
+    # Union{<:Rotation{2},<:RotMatrix{2}} seems able to be replaced with Rotation{2},
+    # but it avoids the method for general dimensional RotMatrix.
```

Comment:
I don't quite understand this comment

Reply:
We can produce random `RotMatrix{2}` instance in two ways:

* Generate random `Angle2d` instance and convert it to `RotMatrix{2}`.
* Generate random matrix from `randn(2,2)` and convert it to `RotMatrix{2}` with `nearest_rotation`.

The first one is much effective in terms of performance, and the last one can be generalized to high dimensional rotations.

The type `R <: Union{<:Rotation{2},<:RotMatrix{2}}` in the first line has been changed from the original [`R <: Rotation{2}` ](https://github.com/JuliaGeometry/Rotations.jl/blob/v1.2.0/src/core_types.jl#L43)to be calculated using `Angle2d`.
I was trying to comment on the reason for this change, but maybe I could not make my English understood... :face_with_spiral_eyes:

Reply:
Oh, I think I see. So the point here is that including `<:RotMatrix{2}` makes this method of `rand` *more specific* than the method for general `RotMatrix`?

In that case I'd word this as something like

```julia
# We use the awkward Union{<:Rotation{2},<:RotMatrix{2}} here rather than `Rotation{2}`
# to make this efficient 2D method more specific than the method for general `RotMatrix`
```

Your English is quite good! It's hard to express this tricky technical point succinctly.

Reply:
Thanks for the suggestion! I've updated the comments that way.